Feature: Implementacion de store de clientes

JesusDelmoral7 opened the pull request:
Se agrega el store de clientes y se modifica el achivo de migraciones

JesusDelmoral7 committed:
Feature: Implementacion de store de clientes

Se agrega el store de clientes y se modifica el achivo de migraciones

xV0lk reviewed (changes requested):
Ayudame a revisar estos cambios porfa

JesusDelmoral7 committed:
cambios solucionados

JesusDelmoral7 committed:
Modifica 00008_clients.sql y elimina zibutatu

JesusDelmoral7 committed:
nuevo commit con los cambios completos

xV0lk commented on internal/db/clients_store.go:
```diff
@@ -0,0 +1,111 @@
+
+	return client, nil
+
+}
```
La función debe cumplir su objetivo de entregar un solo cliente.

---- TODO: -----

Implementar un método que te permita obtener varios clientes y que pueda recibir criterios de búsqueda como:

nombre, teléfono, emai y notificaciones

xV0lk commented on internal/db/clients_store.go:
```diff
@@ -0,0 +1,111 @@
+
+	var exists int
+
+	err := s.db.QueryRow(ctx, queryCheck, client.ID, client.StudioID).Scan(&exists)
```
Esta validación no es necesaria.

Estás tratando de fitrar por un id que aún no ha sido asignado por la bd

xV0lk commented on internal/db/clients_store.go:
```diff
@@ -0,0 +1,111 @@
+
+	return client, nil
+
+}
```
los valores de client.Name, ... son nil ya que no se está recibiendo el objeto client con la data necesaria

xV0lk reviewed (changes requested):
revisar

Commit:
"feat: corrige errores en el store, agrega handlers de clientes y actualiza rutas

- Se corrigen errores en el store relacionados con la obtención de clientes.
- Se implementan nuevos handlers para la gestión de clientes.
- Se actualizan las rutas en el archivo main.go para integrar los nuevos handlers."

Commit:
se agrega handler

Commit:
se modifica archivo de migracion de clients

Commit:
correccion de client sql

Commit:
fix: Se corrigen los handlers y stores

Se corrigen los handlers para recibir los datos correctamente en las funciones de updateclient y getclients
Se corrigen los comandos sql del clientstore para funcionar correctamente
Se corrige el modelo de clientes (especificamente el valor studio_id para que sea reconocido como json)

Commit:
feat: Se agregan handlers y stores de studios

= Se agregan nuevos handlers para para los estudios
= Se agrega nueva store para los estudios
= Se agregan las nuevas rutas al archivo main.go

Commit:
fix: Se corrige la funcion createstudio de la store de studios

Commit:
Se agrega funcionalidad para poder buscar studios por varios parametros, como email, nombre y correo
